src/viam/sdk/common/proto_convert_vector.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <vector>
+
+#include <boost/range/iterator_range.hpp>
+
+#include <viam/sdk/common/proto_convert.hpp>
+
+namespace google {
+namespace protobuf {
+
+template <typename T>
+class RepeatedPtrField;
+
+}
+}  // namespace google
+
+namespace viam {
+namespace sdk {
+namespace proto_convert_details {
+
+template <typename SdkType>
+struct to_proto<std::vector<SdkType>> {
+    void operator()(
+        const std::vector<SdkType>& vec,
+        ::google::protobuf::RepeatedPtrField<EquivalentApiType<SdkType>>* ptr_field) const {
+        ptr_field->Reserve(vec.size());
+        for (const auto& elem : vec) {
+            to_proto<SdkType>{}(elem, ptr_field->Add());
+        }
+    }
+};
+
+template <typename ApiType>
+struct from_proto<::google::protobuf::RepeatedPtrField<ApiType>> {
```

Comment:
I'm not totally sold on making these part of our public API. Or, at least not yet. Could these be in `common/private` and under `viam::sdk::impl`?

I don't doubt the utility, but I'm not sure we need to be in the position of exposing helpers like this for the cases where consumers have opted into dealing with our low-level proto stuff.

It is easier to keep it private now and expose it if it proves worthwhile, than to take it away later.

Reply:
I think that's a better call, I was actually worrying myself about this today because I think there may be cases where it might be `RepeatedField` vs `RepeatedPtrField` and this would be an annoying API lock-in

Reply:
OK, I think let's move them to internal then, since it seems we have similar thoughts here.

Reply:
apparently the choice of repeated field vs ptr field is actually categorical: https://protobuf.dev/reference/cpp/api-docs/google.protobuf.repeated_field/#

but for now I will still leave it in a private header and we can think of `public`izing it if there seems to be a need